roles/ssp/tasks/configure-common.yml

```diff
@@ -60,18 +95,56 @@
   tags:
     - ssp:config:authsources
 
+
 - name: Generate self-signed IdP certificates
   command: openssl req -newkey rsa:2048 -new -x509 -days 3652 -subj "/CN={{ item.ssl_certificate_cn }}" -nodes -out idp-{{ item.entity_id }}.crt -keyout idp-{{ item.entity_id }}.key
   args:
     chdir: "{{ ssp_certdir }}"
     creates: "{{ ssp_certdir }}/idp-{{ item.entity_id }}.key"
   with_items: "{{ ssp_idp_hosts }}"
-  when: ssp_idp_saml20_enabled or ssp_idp_shib13_enabled
+  when: (ssp_idp_saml20_enabled or ssp_idp_shib13_enabled) and item.generate_certificate
   become: yes
   tags:
     - ssp:config:saml20-idp-hosted
     - ssp:config:shib13-idp-hosted
 
+
+
+
+- name: Copy IdP certificate (.crt)
+  copy:
+    content: '{{ item.idp_ssl_certificate }}'
+    dest: "{{ ssp_certdir }}/idp-__DYNAMIC:1__.crt"
+    owner: "{{ ssp_webuser }}"
+    group: "{{ ssp_webgroup }}"
+    mode: "0644"
+  with_items: "{{ ssp_idp_hosts }}"
+  when: item.generate_certificate|bool == False
+  become: yes
+  no_log: yes
+  tags:
+    - ssp:config:saml20-idp-hosted
+    - ssp:config:shib13-idp-hosted
+
+
+- name: Copy IdP certificate (.key)
+  copy:
+    content: '{{ item.idp_ssl_certificate }}'
+    dest: "{{ ssp_certdir }}/idp-__DYNAMIC:1__.key"
```

Comment:
Change `idp-__DYNAMIC:1__.key` to `idp-{{ item.entity_id }.key`. See also:
https://github.com/rciam/rciam-deploy/blob/5e4c2b009e4c6d7a5db3becd0f8584a2c9497a8b/roles/ssp/tasks/configure-common.yml#L64